app/src/main/java/org/greenstand/android/TreeTracker/fragments/LoginFragment.kt

```diff
@@ -0,0 +1,182 @@
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                inactivateLoginButton()
+            }
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                inactivateLoginButton()
+            }
+
+            override fun afterTextChanged(editTextInputs: Editable) {
+                phoneNumberEntered = editTextInputs.toString()
+                viewModel.isUserPresentOnDevice(phoneNumberEntered!!).observe(this@LoginFragment,
+                    Observer {
+                        if (it != null && it.identifier == phoneNumberEntered) {
+                            activateLoginButton()
+                        } else {
+                            sign_up_button.visibility = View.VISIBLE
+                            emailEditText.addTextChangedListener(object : TextWatcher {
+                                override fun afterTextChanged(editTextInputs: Editable?) {
+                                    emailEntered = editTextInputs.toString()
+                                    viewModel.isUserPresentOnDevice(emailEntered!!).observe(this@LoginFragment,
+                                        Observer {
+                                            if (it != null && (it.identifier == emailEntered)) {
+                                                activateLoginButton()
+                                            } else {
+                                                inactivateLoginButton()
+                                                sign_up_button.visibility = View.VISIBLE
+                                            }
+                                        })
+                                }
+                                override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                                    inactivateLoginButton()
+                                }
+                                override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                                    inactivateLoginButton()
+                                }
+                            })
+                        }
+                    })
+            }
+        })
+        sign_up_button.setOnClickListener{
+            val termsFragment = TermsPolicyFragment()
+            val extras = Bundle()
+            extras.apply {
+                putString(PHONE_NUMBER, phoneNumberEntered)
+                putString(EMAIL_ADDRESS, emailEntered)}
+            termsFragment.arguments = extras
+            val fragmentTransaction = activity?.supportFragmentManager?.beginTransaction()
+            fragmentTransaction?.addToBackStack(null)?.replace(R.id.containerFragment, termsFragment)
+            fragmentTransaction?.commit()
+
+        }
+    }
+@SuppressLint("NewApi")
+fun activateLoginButton(){
+    login_button.apply {
+        setTextAppearance(R.style.ActiveButtonStyle)
```

Comment:
There is a way to set enabled/disables styles on a view in xml, then you'd only have to say login_button.enabled = true/false